geo/README.md

```diff
@@ -1,5 +1,273 @@
+
+```
+{
+ "type": "multilinestring",
+ "coordinates": [
+    [ [ -118.31726074, 35.250105158],[ -117.509765624, 35.3756141] ],
+    [ [ -118.6962890, 34.624167789],[ -118.317260742, 35.03899204] ],
+    [ [ -117.9492187, 35.146862906], [ -117.6745605, 34.41144164] ]
+]
+}
+```
+
+## MultiPolygon
+
+The following specifies a [MultiPolygon](https://tools.ietf.org/html/rfc7946#section-3.1.7) field in a document:
+
+```
+{
+ "type": "multipolygon",
+ "coordinates": [
+    [ [ [ -73.958, 40.8003 ], [ -73.9498, 40.7968 ], 
+        [ -73.9737, 40.7648 ], [ -73.9814, 40.7681 ], 
+        [ -73.958, 40.8003 ] ] ],
+
+
+    [ [ [ -73.958, 40.8003 ], [ -73.9498, 40.7968 ], 
+        [ -73.9737, 40.7648 ], [ -73.958, 40.8003 ] ] ]
+ ]
+}
+```
+
+
+## GeometryCollection
+
+The following specifies a [GeometryCollection](https://tools.ietf.org/html/rfc7946#section-3.1.8) field in a document:
+
+```
+{
+ "type": "geometrycollection",
+ "geometries": [
+    {
+      "type": "multipoint",
+      "coordinates": [
+         [ -73.9580, 40.8003 ],
+         [ -73.9498, 40.7968 ],
+         [ -73.9737, 40.7648 ],
+         [ -73.9814, 40.7681 ]
+      ]
+    },
+    {
+      "type": "multilinestring",
+      "coordinates": [
+         [ [ -73.96943, 40.78519 ], [ -73.96082, 40.78095 ] ],
+         [ [ -73.96415, 40.79229 ], [ -73.95544, 40.78854 ] ],
+         [ [ -73.97162, 40.78205 ], [ -73.96374, 40.77715 ] ],
+         [ [ -73.97880, 40.77247 ], [ -73.97036, 40.76811 ] ]
+      ]
+    },
+    {
+      "type" : "polygon",
+      "coordinates" : [
+    [ [ 0 , 0 ] , [ 3 , 6 ] , [ 6 , 1 ] , [ 0 , 0 ] ],
+    [ [ 2 , 2 ] , [ 3 , 3 ] , [ 4 , 2 ] , [ 2 , 2 ] ]
+    ]
+   }
+]
+}
+```
+
+
+## Circle
+
+If the user wishes to cover a circular region over earth’s surface, then they could use this shape.
+A  sample circular shape is as below.
+
+```
+{ 
+ "type": "circle", 
+ "coordinates": [75.05687713623047,22.53539059204079],
+ "radius": "1000m"
+}
+```
+
+ 
+Circle is specified over the center point coordinates along with the radius.
+Example formats supported for radius are: 
+"5in" , "5inch" , "7yd" , "7yards",  "9ft" , "9feet", "11km", "11kilometers", "3nm" 
+"3nauticalmiles", "13mm" , "13millimeters",  "15cm", "15centimeters", "17mi", "17miles" "19m" or "19meters".
+
+If the unit cannot be determined, the entire string is parsed and the unit of meters is assumed.
+
+
+## Envelope
+
+Envelope type, which consists of coordinates for upper left and lower right points of the shape 
+to represent a bounding rectangle in the format  [[minLon, maxLat], [maxLon, minLat]].
+
+```
+{
+    "type": "envelope",
+    "coordinates": [
+      [72.83, 18.979],
+      [78.508,17.4555]
+    ]
+}
+```
+
+
+## GeoShape Query
+
+Geoshape query support three types/filters of spatial querying capability across those 
+heterogeneous types of documents indexed.
+
+###<<>> Query Structure:
```

Comment:
This particular heading's rendering is a bit off, should the "<<>>" part should be removed?